pkg/splunk/enterprise/util.go

```diff
@@ -47,11 +49,34 @@ func ApplySplunkConfig(client splcommon.ControllerClient, cr splcommon.MetaObjec
 		if err = splctrl.ApplyConfigMap(client, defaultsMap); err != nil {
 			return nil, err
 		}
+		// We will update the annotation for resource version in the pod template spec
+		// of the CR, so that any change in the ConfigMap will lead to recycle of the pod.
+		err = updateDefaultConfigMapAnnotations(client, cr, defaultsMap)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	return namespaceScopedSecret, nil
 }
 
+// updateDefaultConfigMapAnnotations checks and gets the default config map and updates the annotation
+// for the resource version of the ConfigMap in the pod template spec for the CR.
+func updateDefaultConfigMapAnnotations(client splcommon.ControllerClient, cr splcommon.MetaObject, configMap *corev1.ConfigMap) error {
+	namespacedName := types.NamespacedName{Namespace: configMap.GetNamespace(), Name: configMap.GetName()}
+	var current corev1.ConfigMap
+
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		return err
+	}
+	annotationsMap := cr.GetObjectMeta().GetAnnotations()
+	if annotationsMap != nil {
```

Comment:
Also, if the map is not allocated, we need to allocate one, and add the entry

Reply:
+1

Reply:
+1